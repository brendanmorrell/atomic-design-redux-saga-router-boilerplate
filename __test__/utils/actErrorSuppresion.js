// temporary patch until act error is fixed in react alpha 16.9.1
let originalError = console.error;
beforeEach(
  () =>
    (console.error = (...args) => {
      originalError(
        ...args.filter(arg => typeof arg === 'string' && !arg.includes('act(')),
      );
    }),
);

afterEach(() => (console.error = originalError));
