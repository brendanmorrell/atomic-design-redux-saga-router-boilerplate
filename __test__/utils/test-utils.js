import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import theme from '../../src/theme';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { storeFactory, history } from '../../src/store';

export const store = storeFactory({});
const customRender = (node, options) =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>{node}</ThemeProvider>
      </Router>
    </Provider>,
    options,
  );

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
