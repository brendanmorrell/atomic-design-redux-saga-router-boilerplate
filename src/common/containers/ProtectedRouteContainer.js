import React from 'react'
import { Route, Redirect } from 'react-router'
import { connect } from 'react-redux'

const ProtectedRoute = ({ loggedIn, component, ...props }) =>
  loggedIn ? <Route component={component} {...props} /> : <Redirect to="/login" />

const mstp = state => ({
  loggedIn: state.auth.loggedIn,
})

export default connect(mstp)(ProtectedRoute)
