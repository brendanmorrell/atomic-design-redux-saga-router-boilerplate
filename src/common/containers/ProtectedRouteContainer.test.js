import React from 'react';
import { render } from '../../../__test__/utils/test-utils';
import ProtectedRouteContainer from './ProtectedRouteContainer';
describe('<ProtectedRouteContainer />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<ProtectedRouteContainer />);
    expect(asFragment).toMatchSnapshot();
  });
});
