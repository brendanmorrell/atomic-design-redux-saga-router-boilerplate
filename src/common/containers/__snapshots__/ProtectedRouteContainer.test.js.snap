// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRouteContainer /> should render correctly 1`] = `[Function]`;
